public/components/register_model/__tests__/register_model_form.test.tsx

```diff
@@ -24,6 +24,7 @@ jest.mock('../../../../../../src/plugins/opensearch_dashboards_react/public', ()
     ...jest.requireActual('../../../../../../src/plugins/opensearch_dashboards_react/public'),
   };
 });
+jest.mock('../../../../public/apis/model');
```

Comment:
Is this `jest.mock` necessary? Since there is already `jest.spyOn(Model.prototype, 'search')`

Reply:
If remove this `jest.mock` and the two `jest.mock` in `setup.tsx`, the tests will be running good.

I think we should avoid writing tests that implicitly rely on mocks from other places. Define `jest.mock` in `setup.ts` and importing it in test files is generally not a good practice due to the way how `jest.mock` [works](https://jestjs.io/docs/manual-mocks#using-with-es-module-imports):
> Jest will automatically hoist `jest.mock` calls to the top of the module (before any imports)

To write robust tests, I found these rules are useful
1. Defined `jest.mock` at where it's used. This makes the test suit self-contained
2. Defined mocks such as `spyOn` at where it's used, try to NOT define those in `beforeEach`. This makes the unit test self-contained.